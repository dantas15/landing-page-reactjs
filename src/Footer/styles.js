import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
