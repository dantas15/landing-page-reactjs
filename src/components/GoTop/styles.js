import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;

    bottom: 2rem;
    right: 2rem;

    z-index: 6;

    border-radius: 0.4rem;

    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
  `}
`;
