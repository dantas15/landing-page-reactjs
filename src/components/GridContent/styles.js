import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    margin: 0 auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge} 0;
  `}
`;
