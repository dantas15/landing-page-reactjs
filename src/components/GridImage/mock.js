export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      altText: 'Teste 1',
      srcImage: 'https://source.unsplash.com/random/800x800?r=1',
    },
    {
      altText: 'Teste 2',
      srcImage: 'https://source.unsplash.com/random/800x800?r=2',
    },
    {
      altText: 'Teste 3',
      srcImage: 'https://source.unsplash.com/random/800x800?r=3',
    },
    {
      altText: 'Teste 4',
      srcImage: 'https://source.unsplash.com/random/800x800?r=4',
    },
    {
      altText: 'Teste 5',
      srcImage: 'https://source.unsplash.com/random/800x800?r=5',
    },
    {
      altText: 'Teste 6',
      srcImage: 'https://source.unsplash.com/random/800x800?r=6',
    },
  ],
};
