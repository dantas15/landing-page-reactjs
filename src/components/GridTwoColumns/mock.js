export default {
  title: 'Grid Two Columns',
  text: `Lorem ipsum dolor sit, amet consectetur adipisicing elit.
  Repellendus, voluptatibus? Fuga aut et illum voluptates neque
  molestiae consequatur, voluptatibus enim consequuntur delectus
  sed unde ratione velit deleniti eligendi, culpa ipsa?`,
  srcImg: 'assets/images/javascript.svg',
};
