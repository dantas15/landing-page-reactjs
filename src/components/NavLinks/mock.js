export default [
  {
    children: 'Link 1',
    link: '#target1',
    newTab: false,
  },
  {
    children: 'Link 2',
    link: '#target2',
    newTab: false,
  },
  {
    children: 'Link 3',
    link: '#target3',
    newTab: false,
  },
  {
    children: 'Link 4',
    link: '#target4',
    newTab: false,
  },
  {
    children: 'Link 5',
    link: '#target5',
    newTab: false,
  },
];
