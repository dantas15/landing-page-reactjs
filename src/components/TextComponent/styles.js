import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
  `}
`;
