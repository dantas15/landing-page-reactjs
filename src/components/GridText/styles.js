import styled, { css } from 'styled-components';

import { Title as HeadingContainer } from 'components/Heading/styles';
import { Container as TextComponent } from 'components/TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;

    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.xhuge};
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    } ;
  `}
`;
