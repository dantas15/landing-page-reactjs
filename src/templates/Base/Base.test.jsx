import { screen } from '@testing-library/react';

import { renderTheme } from 'styles/render-theme';
import { Base } from '.';

import { mockBase } from './mock';

describe('<Base />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Base {...mockBase} />);
    expect(container).toMatchSnapshot();
  });
});
