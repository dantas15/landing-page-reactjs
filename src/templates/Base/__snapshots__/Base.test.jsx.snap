// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> should render 1`] = `
.c11 {
  padding-top: 5.4rem;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c7 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c14 {
  color: #ffffff;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c19 {
  color: #ffffff;
  font-size: 3.2rem;
  text-transform: none;
}

.c21 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c22 {
  color: #0A1128;
  font-size: 3.2rem;
  text-transform: none;
}

.c2 {
  position: fixed;
  z-index: 5;
  top: 0;
  right: 0;
  left: 0;
  width: 100%;
  border-bottom: #DDDDDD;
  background: #ffffff;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 > .c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c2 .c6 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background: #0A1128;
  color: #ffffff;
  border: none;
  display: none;
  pointer-events: all;
}

.c0 > svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c8 > img {
  max-height: 3rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c10 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
  position: relative;
}

.c10::after {
  content: '';
  position: absolute;
  bottom: 0.76rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #DC143C;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c10:hover::after {
  left: 25%;
  width: 50%;
}

.c16 {
  font-size: 2.4rem;
}

.c23 {
  text-align: center;
  border-top: 0.1rem solid #DDDDDD;
}

.c23 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c23 .c15 {
  font-size: 1.6rem;
}

.c23 .c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c24 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
  border-radius: 0.4rem;
  background: #0A1128;
  color: #ffffff;
}

.c13 .c15 {
  margin-bottom: 6.4rem;
}

.c17 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 3.2rem;
}

.c18 .c6 {
  position: relative;
  left: 5rem;
}

.c18 .c6::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 6.4rem;
  top: -3rem;
  left: -5rem;
  -webkit-transform: rotate(5deg);
  -ms-transform: rotate(5deg);
  transform: rotate(5deg);
}

.c12 {
  background: #ffffff;
  color: #0A1128;
  background: #0A1128;
  color: #ffffff;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c20 {
  background: #ffffff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:768px) {
  .c11 {
    padding-top: 0;
  }
}

@media (max-width:768px) {
  .c14 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c21 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    height: 100vh;
    visibility: hidden;
    opacity: 0;
  }

  .c2 > .c3 {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100vh;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    overflow-y: auto;
  }

  .c2 .c6 {
    padding-bottom: 3.2rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c5 {
    display: block;
    text-align: center;
    padding: 4.8rem 0;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c9 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

@media (max-width:768px) {
  .c17 {
    grid-template-columns: 1fr;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6 c7"
        >
          <a
            class="c8"
            href="#"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c9"
        >
          <a
            class="c10"
            href="#target1"
            target="_self"
          >
            Link 1
          </a>
          <a
            class="c10"
            href="#target2"
            target="_self"
          >
            Link 2
          </a>
          <a
            class="c10"
            href="#target3"
            target="_self"
          >
            Link 3
          </a>
          <a
            class="c10"
            href="#target4"
            target="_self"
          >
            Link 4
          </a>
          <a
            class="c10"
            href="#target5"
            target="_self"
          >
            Link 5
          </a>
          <a
            class="c10"
            href="#target6"
            target="_self"
          >
            Link 6
          </a>
          <a
            class="c10"
            href="#target7"
            target="_self"
          >
            Link 7
          </a>
          <a
            class="c10"
            href="#target8"
            target="_self"
          >
            Link 8
          </a>
          <a
            class="c10"
            href="#target9"
            target="_self"
          >
            Link 9
          </a>
          <a
            class="c10"
            href="#target10"
            target="_self"
          >
            Link 10
          </a>
          <a
            class="c10"
            href="#target11"
            target="_self"
          >
            Link 11
          </a>
          <a
            class="c10"
            href="#target12"
            target="_self"
          >
            Link 12
          </a>
        </nav>
      </div>
    </div>
  </div>
  <div
    class="c11"
  >
    <div
      class="c12"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h2
            class="c6 c14"
          >
            My grid
          </h2>
          <footer
            class="c15 c16"
          >
            Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
          </footer>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                Teste 1
              </h3>
              <footer
                class="c15 c16"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </footer>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                Teste 2
              </h3>
              <footer
                class="c15 c16"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </footer>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                Teste 3
              </h3>
              <footer
                class="c15 c16"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </footer>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c20"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h2
            class="c6 c21"
          >
            My grid
          </h2>
          <footer
            class="c15 c16"
          >
            Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
          </footer>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                Teste 1
              </h3>
              <footer
                class="c15 c16"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </footer>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                Teste 2
              </h3>
              <footer
                class="c15 c16"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </footer>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                Teste 3
              </h3>
              <footer
                class="c15 c16"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </footer>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h2
            class="c6 c14"
          >
            My grid
          </h2>
          <footer
            class="c15 c16"
          >
            Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
          </footer>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                Teste 1
              </h3>
              <footer
                class="c15 c16"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </footer>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                Teste 2
              </h3>
              <footer
                class="c15 c16"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </footer>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c19"
              >
                Teste 3
              </h3>
              <footer
                class="c15 c16"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </footer>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c20"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h2
            class="c6 c21"
          >
            My grid
          </h2>
          <footer
            class="c15 c16"
          >
            Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
          </footer>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                Teste 1
              </h3>
              <footer
                class="c15 c16"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </footer>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                Teste 2
              </h3>
              <footer
                class="c15 c16"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </footer>
            </div>
            <div
              class="c18"
            >
              <h3
                class="c6 c22"
              >
                Teste 3
              </h3>
              <footer
                class="c15 c16"
              >
                Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.
              </footer>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c23"
    >
      <div
        class="c3 c4"
      >
        <footer
          class="c15 c16"
        >
          <p>
            Teste de footer
          </p>
        </footer>
      </div>
    </div>
  </div>
  <a
    aria-label="Go to top"
    class="c24"
    href="#"
    title="Go to top"
  >
    <svg
      aria-hidden="true"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
      />
    </svg>
  </a>
</div>
`;
